src/common/src/types/mod.rs

```diff
@@ -691,6 +697,139 @@ impl ScalarImpl {
             }
         })
     }
+
+    pub fn to_protobuf(d: &Datum) -> Vec<u8> {
```

Comment:
We can define a trait called ToProtobuf and let Datum impl it so that it will be able to;

```rust
impl ToProtobufBytes for Datum {
  fn to_protobuf_bytes(&self) -> Vec<u8> {
  }
}
```

```rust
let d: Datum = ...;
d.to_protobuf();
```